utility/RTC_Super.h

```diff
@@ -0,0 +1,74 @@
+/**************************************************************************/
+/*!
+  @file     RTC_Super.h
+
+  Original library by JeeLabs http://news.jeelabs.org/code/, released to the
+  public domain
+
+  License: MIT (see LICENSE)
+
+  This is a fork of JeeLab's fantastic real time clock library for Arduino.
+
+  For details on using this library with an RTC module like the DS1307, PCF8523,
+  or DS3231, see the guide at:
+  https://learn.adafruit.com/ds1307-real-time-clock-breakout-board-kit/overview
+
+  Adafruit invests time and resources providing this open source code,
+  please support Adafruit and open-source hardware by purchasing
+  products from Adafruit!
+*/
+/**************************************************************************/
+
+#ifndef _RTC_SUPER_H_
+#define _RTC_SUPER_H_
+
+#include "DateTime.h"
+#include <Arduino.h>
+
+/**************************************************************************/
+/*!
+    @brief  RTC superclass for all RTC chips
+*/
+/**************************************************************************/
+class RTC_Super {
+public:
+  /*!
+    @brief Start the RTC while preserving the RTC's date/time
+    @return True if successful, false otherwise
+  */
+  virtual boolean begin(void) = 0;
+
+  /*!
+    @brief Start the RTC and adjust the RTC's date/time
+    @param dt DateTime object containing desired date/time
+    @return True if successful, false otherwise
+  */
+  virtual boolean begin(const DateTime &dt) = 0;
```

Comment:
This was done so that all RTC classes have both a `begin(void)` and `begin(DateTime&)` methods, as opposed to some only having the former and others the latter. Making it so that the `begin()` methods in the superclass are not pure virtual defeats the point of having a superclass.

Reply:
Maybe `begin(void)` could be the only version that is mandatory. The “soft” RTCs would provide a simplistic implementation (say, setting the time to 2000-01-01T00:00:00) and provide `begin(DateTime&)` for backward compatibility. The soft RTCs are the only ones where _not_ updating the RTC on startup makes little sense.

Reply:
I am not sure I understand that. There is nothing that can be done if the derived extends the base class. The base class is constructed as an interface and should be abstract. The main appeal is polymorphism. 

Begin is an interesting notion. Overall I see your point. However I think this could cause confusion or misusage. I see little difference between calling begin and adjust in sequence. However it is technically bad forum. However begin is technically bad form.

Personally there is an argument for dropping both begins. There are cases where it is useful like serial or async protocols which can become receivers once enabled. For binary counters or software notions I could see the argument for a begin function existing.

Or perhaps you keep both and have then do nothing. Which would prevent hammering the battery backed versions. The question is where does the seed value come from, but there is other usages for RTCs like alarm functions where this may not matter.